middleware.go

```diff
@@ -77,7 +77,7 @@ func parseRequestURL(c *Client, r *Request) error {
 }
 
 func parseRequestHeader(c *Client, r *Request) error {
-	hdr := http.Header{}
+	var hdr = make(http.Header, len(c.Header))
```

Comment:
This change may not be beneficial. Since this `len(c.Header)` is not actual count of the total headers. 
Total headers => Client headers + Reader headers - Request Headers override

Also it terms performance, there is no noticeable difference in general for resty. Also just choosing only `len(c.Header)` is not helping us.

Reply:
Yeah, I agree.